### PR TITLE
[VSL-24] Fix code coverage

### LINKED ISSUE #24: Code test coverage reporting as 0%
It looks like a bug with AGP 7 is causing Jacoco to report a code coverage of 0%: https://issuetracker.google.com/issues/195860510

### PR DESCRIPTION
**Pull request by malachid**
JaCoCo has been broken since upgrading to JDK11.
For now, at least, changing over to Kover + IntelliJ to get coverage reporting working.

Resolves #24

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -15,6 +15,7 @@ buildscript {
 plugins {
     id("com.vanniktech.android.junit.jacoco") version Versions.vannikJacoco
```
Comment (a-ware) [inline build.gradle.kts]: Shouldn't need the jacoco plugin anymore

File: buildSrc/src/main/kotlin/Versions.kt [hunk 2]
```diff
@@ -27,6 +27,8 @@ object Versions {
     const val vannikJacoco = "0.16.0"
     const val jacoco = "0.8.7"
```
Comment (a-ware) [inline buildSrc/src/main/kotlin/Versions.kt]: Also a good time to clean up the jacoco dependencies

File: build.gradle.kts [hunk 3]
```diff
@@ -25,6 +26,17 @@ allprojects {
     }
 }
 
+kover {
+    isDisabled = false                                   // true to disable instrumentation of all test tasks in all projects
+    coverageEngine.set(kotlinx.kover.api.CoverageEngine.INTELLIJ) // change instrumentation agent and reporter
+    intellijEngineVersion.set(Versions.intellijCover)    // change version of IntelliJ agent and reporter
+    jacocoEngineVersion.set(Versions.jacoco)             // change version of JaCoCo agent and reporter
```
Comment (a-ware) [inline build.gradle.kts]: And since we don't plan to use jacaco, this line could be excluded

**Review by a-ware — COMMENTED** [review]
Looks good and works for me locally, but some more jacaco clean up we could do

**Comment by malachid** [comment]
> Looks good and works for me locally, but some more jacaco clean up we could do

sure I'll go ahead and remove the jacoco stuff

**Review by a-ware — APPROVED** [review]

**Review by tpearson — APPROVED** [review]

**Pull request merged**